apps/etterlatte-behandling/src/main/kotlin/oppgaveny/OppgaveDaoNy.kt

```diff
@@ -77,14 +78,45 @@ class OppgaveDaoNy(private val connection: () -> Connection) {
         }
     }
 
-    fun hentOppgaver(): List<OppgaveNy> {
+    fun hentOppgaver(oppgaveTyper: List<OppgaveType>): List<OppgaveNy> {
+        if (oppgaveTyper.isEmpty()) return emptyList()
+
         with(connection()) {
             val statement = prepareStatement(
                 """
-                    SELECT id, status, enhet, sak_id, type, saksbehandler, referanse, merknad, opprettet, saktype, fnr, frist
-                    FROM oppgave
+                    SELECT o.id, o.status, o.enhet, o.sak_id, o.type, o.saksbehandler, o.referanse, o.merknad, o.opprettet, o.saktype, o.fnr, o.frist
+                    FROM oppgave o INNER JOIN sak s ON o.sak_id = s.id
+                    WHERE o.type = ANY(?)
+                    AND s.adressebeskyttelse is null OR 
+                    (s.adressebeskyttelse is NOT NULL AND (s.adressebeskyttelse != ? AND s.adressebeskyttelse != ?))
+                """.trimIndent()
+            )
+            statement.setArray(1, createArrayOf("text", oppgaveTyper.toTypedArray()))
+            statement.setString(2, AdressebeskyttelseGradering.STRENGT_FORTROLIG.toString())
```

Comment:
Kanksje preferanse-sak, men eg kjenner meg meir trygg på å bruke .name, toString() er det ein risiko for at vi overstyrer

Reply:
https://github.com/navikt/pensjon-etterlatte-saksbehandling/pull/1842/commits/ac671289ca8456f3e7dbb35c51096f0823c6f077